### ISSUE TITLE
Addon settings are not restored on reinstallation

### ISSUE THREAD

**Issue by donovaly** [yorikvanhavre/FreeCAD-Help#14]
Assume the user has e.g. set this:
![FreeCAD_8nYRUNu0lJ](https://user-images.githubusercontent.com/1828501/174079646-a3b81de3-1a56-4ffa-9db2-f338bb326db1.png)

- He notices that the What's This? feature does not work and decides to uninstall the Help Addon using the Addon Manager

- then he restarts FC and installs the help addon again hoping to get it fixed

result: he ends up with the same broken settings he has before.

I am not sure this is a bug in either the help Addon, the AddonManager or somewhere in FreeCAD. @chennes , could you please have a look?

**Comment by donovaly** [comment]
Also note that ALL settings are kept on uninstalling. So an uninstallation does not bring your system back to the default.

**Comment by chennes** [comment]
I suggest adding an "uninstall.py" file to this addon that removes those settings. That should get run automatically by the AM when you hit uninstall.

**Comment by yorikvanhavre** [comment]
Oh that's useful. But in this case, do we WANT that? I mean, do we want the uninstalling of an addon to clean its settings? That would be similar to, on linux, if you remove a package, it would also remove your personal settings. Most people would be against such behaviour.

In the example here, of course it makes sense if the user does something wrong and would want to undo it. But what about other settings? What if you set, for example, the URL of a doc server in your own company? You would want it still stored if you need to uninstall/reinstall the addon

**Comment by donovaly** [comment]
> I mean, do we want the uninstalling of an addon to clean its settings?

Yes, definitely. For the Windows installer I spent much time to leave a clean system after the uninstallation. This is important also for many cases where users somehow mangled their system. Now they can uninstall FreeCAD and on reinstalling get a working system back. The uninstaller therefore has an option to delete also the user preferences. So the user can choose between keeping the settings ore not.

However, for the Addons I don't think it is worth to keep the user settings. Usually when uninstalling an Addon you either don't want to use it, or you want a reinstall to refresh it.

Thinking about it, I cannot imagine a use case where I want to uninstall an addon and ten reinstall it later with the same settings. Do you know such a use case?

> What if you set, for example, the URL of a doc server in your own company? You would want it still stored if you need to uninstall/reinstall the addon

But why do I uninstall the addon when it works as it should?
As I argumented, when I uninstall an addon to reinstall it subsequently, I do this to refresh the Addon. When my buggy settings are kept, (and obviously they are buggy, otherwise there is no need to reinstall), then I have no benefit and must fiddle around to find out what of my settings cause the issue I am suffering.

Moreover, when the addon settings are kept, it is also for the forum people hard to help users who somehow mangled their Addon settings.

**Comment by yorikvanhavre** [comment]
In any case the uninstall.py file is a good idea and can be used to solve the above "problem"... It would be easy to either 1) have the addon manager ask the user "do you want to wipe addon settings?" or 2) do that inside the uninstall.py... What do you think would be best @chennes ?

**Comment by yorikvanhavre** [comment]
Ok, let me do it here then, if useful we can always migrate that later. Indeed if we provide workbench authors with a custom uninstall system, it makes also sense to let them decide what to keep, what the user needs to decide, etc.
